src/linux/backend-v4l2.cpp

```diff
@@ -1181,53 +1182,82 @@ namespace librealsense
-                                                // Payload:
-                                                meta_data.report_type.imu_report.header.md_type_id = md_type::META_DATA_HID_IMU_REPORT_ID;
-                                                meta_data.report_type.imu_report.header.md_size = metadata_imu_report_size;
-
-                                                md_size = sizeof(metadata_hid_raw);
-                                                md_start = &meta_data;
+                                                LOG_WARNING("Video frame dropped, video and metadata buffers inconsistency");
                                             }
+                                        }
+                                        else // when metadata is not enabled at all, streaming only video
+                                        {
+                                            auto timestamp = (double)buf.timestamp.tv_sec * 1000.f + (double)buf.timestamp.tv_usec / 1000.f;
+                                            timestamp = monotonic_to_realtime(timestamp);
 
-                                            frame_object fo{ frame_sz, md_size,
-                                                        buffer->get_frame_start(), md_start, timestamp };
+                                            LOG_DEBUG_V4L("no metadata streamed");
+                                            if (buf_mgr.verify_vd_md_sync())
+                                            {
+                                                buffer->attach_buffer(buf);
+                                                buf_mgr.handle_buffer(e_video_buf, -1); // transfer new buffer request to the frame callback
 
-                                            //Invoke user callback and enqueue next frame
-                                            _callback(_profile, fo, [buf_mgr]() mutable {
-                                                buf_mgr.request_next_frame();
-                                            });
 
-                                        }
-                                        else
-                                        {
-                                            LOG_WARNING("Video frame dropped, video and metadata buffers inconsistency");
+                                                auto frame_sz = buf_mgr.md_node_present() ? buf.bytesused :
+                                                                    std::min(buf.bytesused - buf_mgr.metadata_size(),
+                                                                             buffer->get_length_frame_only());
+
+                                                uint8_t md_size = buf_mgr.metadata_size();
+                                                void* md_start = buf_mgr.metadata_start();
+
+                                                // D457 development - hid over uvc - md size for IMU is 64
```

Comment:
Can we get here only with D457?

Reply:
No - but the need to update the IMU frames metadata from UVC is only via D457. Only in this case, we will get into the if statement one line below.